src/registries/npm-registry.js

```diff
@@ -139,7 +138,7 @@ export default class NpmRegistry extends Registry {
         await fs.mkdirp(mirrorLoc);
       }
 
-      defaults(this.config, config);
+      this.config = Object.assign({}, config, this.config);
```

Comment:
It might if you're sharing a config object between registries and relying on outside mutations.